platforms/arm_atsam/bootloaders/md_boot.c

```diff
@@ -38,17 +38,21 @@ void bootloader_jump(void) {
 
     if (!*ver_check) {                   // If check version pointer is NULL, all characters have matched
         *MAGIC_ADDR = BOOTLOADER_MAGIC;  // Set magic number into RAM
-        NVIC_SystemReset();              // Perform system reset
+        keyboard_jump();              // Perform system reset
```

Comment:
Looks like most (but not all) implementations of `bootloader_jump()` now use `keyboard_jump()` instead of performing system reset directly; however, `keyboard_jump()` is weak and therefore can be replaced — but with this implementation there is no way to override just the “restart keyboard” function without also affecting the “reboot to bootloader” function. Is this actually intended? Maybe it would be better to add a separate `mcu_reset()` function and use it in both `bootloader_jump()` and `keyboard_jump()` (although there is probably no point doing that for just `NVIC_SystemReset()`, some MCUs need much more elaborate code there).

Also it might make sense to write something invalid into the bootloader magic location for the “restart keyboard” case, so that the restart happens even if something else happened to be using that RAM location. This would also require having a separate “restart keyboard” function.